skills/timezone.js

```diff
@@ -1,63 +1,97 @@
-/*
+const {findTimeZone, getZonedTime} = require('timezone-support');
+const {formatZonedTime} = require('timezone-support/dist/parse-format');
+const ExpressionList = require('../submodules/Regex/ExpressionList');
+const timeZoneExList = require('../submodules/Regex/timeZoneExList');
 
- This simply hears the time in United States timezones and reports the system time (EST) and adjusts it accordingly
-
-*/
+// Zone Name comes from the npm module timezone-support and pulled from a list listTimeZones.
+// Using zone names will account for Daylights Savings Time
+const zones = {
+  PST: 'America/Los_Angeles',
+  MST: 'America/Phoenix',
+  CST: 'America/Chicago',
+  EST: 'America/New_York',
+  IST: 'Indian/Mahe',
+  GMT: 'Europe/London',
+  CEST: 'Europe/Warsaw',
+  CTC: 'Asia/Shanghai'
+};
 
-module.exports = function(controller) {
-  //listening for time
+module.exports = controller => {
   controller.hears(
-    ['^time (.*)'],
-    'direct_message,direct_mention',
-    (bot, message) => {
-      const timezone = message.match[1];
-      const time = getTimeForTimezone(timezone);
-      bot.createConversation(message, (err, convo) => {
-        if (err) return false;
-        const responseMessage = 'The current time is ' + time + '.';
-        convo.say(responseMessage);
-        convo.activate();
-      });
-    }
-  );
+      [ExpressionList.timezone], 
+      'direct_message,direct_mention', 
+      async (bot, message) => {
+          var tz = message.text.split(' ')[1];
```

Comment:
Use regex to get your match from `message.match`, like in the lookup PR suggestion

Reply:
Also try not to abbreviate variable names, it can get really confusing if this gets more complex. Please use `timezone` or something that easily lets you guess what a var might contain.